### PR TITLE
Fix docker uid/gid + binaryData permissions. Persist postgres db

### LINKED ISSUE #3337: document how to apply migrations in locally running docker-based wK
@georgwiese You had this problem, right? Did you solve it with psql in docker? Or did you work around it? Could you let us know what other usability issues you had? Then we can hopefully make things easier both for you and others :)

- [x] `docker-compose run psql` gives access to the db, but in its shell using `\i` I get `conf/evolutions/reversions/32-scratch-datastores.sql: No such file or directory`
- [x] is the db persistent?
- [x] what’s up with the file permissions? failed to create organization dir. binaryData is owned by root now, so as normal user I can’t put datasets in there either
- [x] can I test changes to this locally without them being in master?
- [x] add commented-out   `# - -Dplay.http.router="noDS.Routes"` to docker-compose

### PR DESCRIPTION
**Pull request by fm3**
### Steps to test:
- follow the changed instructions in the readme

### issues
- fixes #3337 (together with new wiki entry)
(note that applying the evolutions will become easier with #3427)

------
- [x] Updated [changelog](../blob/master/CHANGELOG.md#unreleased)
- ~~[ ] Updated [migration guide](../blob/master/MIGRATIONS.md#unreleased) if applicable~~
- ~~[ ] Updated [documentation](../blob/master/docs) if applicable~~
- ~~[ ] Needs datastore update after deployment~~
- [x] Ready for review

### CODE REVIEW

**Comment by fm3** [comment]
TODO: workaround for https://github.com/avajs/ava/issues/1793 for the now persisted pg database is inaccessible when running the front-end files

File: conf/evolutions/reversions/033-tasktype-recommendedConfiguration.sql [hunk 1]
```diff
@@ -1,7 +1,7 @@
 START TRANSACTION;
 
 DROP VIEW webknossos.taskTypes_;
-ALTER TABLE webknossos.taskTypes DROP COLUMN recommendedConfiguration JSONB;
+ALTER TABLE webknossos.taskTypes DROP COLUMN recommendedConfiguration;
```
Comment (jstriebel) [inline conf/evolutions/reversions/033-tasktype-recommendedConfiguration.sql]: Is this simply a fix, missing in another PR?

Comment (fm3) [reply conf/evolutions/reversions/033-tasktype-recommendedConfiguration.sql]: Exactly, I assume nobody tested the reversion before…

File: docker-compose.yml [hunk 2]
```diff
@@ -169,26 +176,42 @@ services:
       - "./pg/dev-db:/var/lib/postgresql/data/"
 
   psql:
-    extends: postgres
+    extends: postgres-persisted
```
Comment (jstriebel) [inline docker-compose.yml]: I think this should rather extend `postgres` and link `postgres-persisted` and use this link in the command. Same for `drop-db`.

File: docker-compose.yml [hunk 3]
```diff
@@ -160,6 +161,12 @@ services:
       POSTGRES_USER: postgres
       POSTGRES_PASSWORD: postgres
 
+  postgres-persisted:
+    extends:
+      service: postgres
+    volumes:
+      - "./pg/db:/var/lib/postgresql/data/"
+
```
Comment (jstriebel) [inline docker-compose.yml]: Maybe we should also remove the ports for `postgres-dev` to avoid conflicts with local postgres instances?

Comment (fm3) [reply docker-compose.yml]: My understanding is that the `-dev` stuff is for developers who did not install all the native stuff, so there shouldn’t be a problem in this case. Exposing the port gives them the possibility to connect with their favourite postgres client. I don’t have a strong opinion though, as I don’t think anyone is currently using this kind of setup.

Comment (jstriebel) [reply docker-compose.yml]: True, let's leave it like this then.

**Pull request merged**